### ISSUE TITLE
Add Documentation on how to inject/access DBContext or information on scoped services in SignalR

### ISSUE THREAD

**Issue by TrieBr** [dotnet/AspNetCore.Docs#9658]
The documentation is not clear how to inject and work with a DBContext or scoped service in SignalR, which I believe would be a fairly common scenario. Hubs have a `Transient` lifetime. According to the dependency injection documentation it says "Services that use the database context should also use the scoped lifetime". It's not clear how scoped services are handled in signalR. 

I'm currently having an issue resulting in error `A second operation started on this context before a previous operation completed. Any instance members are not guaranteed to be thread safe.` on my DBContext intermittently. I have a feeling this is related to the difference in my `Scoped` DBContext and `Transient` Hub. In a normal HTTP request it's clear that a "Scoped" service is created at the beginning of each http request. However in a websocket situation, when is a Scoped service created? Is the scoped service created when the socket establishes connection and the scope ends when the socket is disconnected? It seems like my problem is due to a Transient Hub using the same scoped service across multiple Hub method calls resulting in the threading error mentioned above.

Note: I've already checked for async method invocations that are missing await expressions using Regex and ReSharper, and have found none. I've never had this much of a problem on normal controllers so I'm led to believe this is an issue with how SignalR handles scoped services which isn't documented.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 1387f556-b4b7-70f9-5bd0-f0bccc597689
* Version Independent ID: fb016cdc-d281-6b5b-7466-2c718e7274b4
* Content: [Use hubs in ASP.NET Core SignalR](https://docs.microsoft.com/en-us/aspnet/core/signalr/hubs?view=aspnetcore-2.1)
* Content Source: [aspnetcore/signalr/hubs.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/signalr/hubs.md)
* Product: **aspnet-core**
* GitHub Login: @tdykstra
* Microsoft Alias: **tdykstra**

**Comment by TrieBr** [comment]
After further review, I can see in the source code that a scope is created per method invocation (https://github.com/aspnet/SignalR/blob/d0bcdf3dd92afba858fca2871535faf4594c81de/src/Microsoft.AspNetCore.SignalR.Core/Internal/DefaultHubDispatcher.cs#L166).

I think you can close this issue while I figure out this is a bug in my code or a bug in signalR

**Comment by anurse** [comment]
Yeah, some docs would certainly be good here. Hubs are indeed transient, a new instance is created for each invocation. Singletons will just be injected as-is. Scoped and Transient services will generally be created new for each invocation. We create a DI Scope prior to each individual invocation (as you just noted), so Scoped and Transient services end up having equivalent behavior.

If you can pull together a project that reproduces the problem we can help look at it and see if there's an issue in your code or ours ;). Feel free to file a bug on https://github.com/aspnet/SignalR once you are able to narrow down a repro!

**Comment by dcarapic** [comment]
I think the scope of the hub should be documented. You can kind of assume it since the documentation states that the Hub is transient but a written documentation statement would be better. I came here googling this exact question as I was unsure if I should be writing scope wrappers in my every hub method.

**Comment by dcarapic** [comment]
I think the **transient** scope .... 
It would be nice if I could edit my comment :(

**Comment by douglasg14b** [comment]
The docs could be much clearer about this...

**Comment by Grizzlly** [comment]
After all, if I want to use a DbContext in my hub, is it ok to get it through the constructor? Or should I add DbContextFactory? In my mind, this also ties to Blazor Server which is based on SignalR. From the documentation, it is advised to use DbContextFactory.

**Comment by Neutrino-Sunset** [comment]
The documentation now states that hubs are transient.

https://learn.microsoft.com/en-gb/aspnet/core/signalr/hubs?view=aspnetcore-6.0#create-and-use-hubs

But unless you manually register them yourself they do not appear to be registered at all.

![image](https://user-images.githubusercontent.com/18445931/217011142-e9f95023-b79d-4e02-9c64-8ce03ac782cf.png)

And if you register them yourself, nothing stops you from registering them with whatever lifetime you like, which presumably is likely to cause problems unless you register them as transient. If the documentation says they are registered as transient, then they should be registered as transient by the system itself.

The current implementation (and documentation) is confusing and fragile.

**Comment by BrennanConroy** [comment]
Don't resolve hubs from DI yourself. They are not intended to be called by user code and are purposefully not added to DI.

It looks like you are wanting to use `IHubContext<T>`, see https://learn.microsoft.com/aspnet/core/signalr/hubcontext?view=aspnetcore-6.0
